src/IR/LowerAst.hs

```diff
@@ -123,16 +123,30 @@ lowerExpr (A.Wait exprs) k =
   I.Prim I.Wait (map (`lowerExpr` id) exprs) (k I.untyped)
 lowerExpr (A.Seq l r) k =
   I.Let [(Nothing, lowerExpr l id)] (lowerExpr r id) (k I.untyped)
-lowerExpr A.Break          k = I.Prim I.Break [] (k I.untyped)
-lowerExpr (A.Return e    ) k = I.Prim I.Return [lowerExpr e id] (k I.untyped)
-lowerExpr (A.IfElse c t e) k = I.Match cond Nothing [tArm, eArm] (k I.untyped)
+lowerExpr A.Break        k = I.Prim I.Break [] (k I.untyped)
+lowerExpr (A.Return e  ) k = I.Prim I.Return [lowerExpr e id] (k I.untyped)
+lowerExpr (A.Match s ps) k = I.Match cond (fmap f ps) (k I.untyped)
+ where
+  cond = lowerExpr s id
+  f (a, b) = (lowerAlt a, lowerExpr b id)
+lowerExpr (A.IfElse c t e) k = I.Match cond [tArm, eArm] (k I.untyped)
  where
   cond = lowerExpr c id
   tArm = (I.AltLit (I.LitBool True), lowerExpr t id)
-  eArm = (I.AltDefault, lowerExpr e id)
+  eArm = (I.AltDefault Nothing, lowerExpr e id)
 lowerExpr (A.OpRegion _ _) _ = error "Should already be desugared"
 lowerExpr A.NoExpr         k = I.Lit I.LitEvent (k I.untyped)
 
+-- | Lower an A.Pat into an I.Alt
+lowerAlt :: A.Pat -> I.Alt
+lowerAlt A.PatWildcard  = I.AltDefault Nothing
+lowerAlt (A.PatId  _  ) = error "I.Alt for A.PatID not implemented yet"
```

Comment:
This can now be directly translated to the IR, using `IR.AltDefault (Just v)` wher `v` is the id in `PatId`.